homeassistant/components/media_player/samsungtv.py

```diff
@@ -171,13 +178,25 @@ def supported_features(self):
 
     def turn_off(self):
         """Turn off media player."""
+        self._start_powering_off_timer()
         if self._config['method'] == 'websocket':
             self.send_key('KEY_POWER')
         else:
             self.send_key('KEY_POWEROFF')
         # Force closing of remote session to provide instant UI feedback
         self.get_remote().close()
 
+    def _start_powering_off_timer(self):
+        def _handle_powering_off_end():
+            self._is_powering_off = False
+
+        self._is_powering_off = True
+        if self._powering_off_timer is not None\
+                and self._powering_off_timer.isAlive():
+            self._powering_off_timer.cancel()
+        self._powering_off_timer = Timer(15, _handle_powering_off_end)
```

Comment:
Use `track_point_in_time` instead from helpers/event.py.

Reply:
```
    def _start_powering_off_timer(self):
        self._is_powering_off = True
        track_point_in_time(self.hass, self._handle_powering_off_end,
                            dt_util.utcnow() + timedelta(seconds=15))

    def _handle_powering_off_end(self):
        self._is_powering_off = False
```
With this code, the _handle_powering_off_end is never called. Do you see what I'm doing wrong?
I've checked, self.hass is not None.
Everything seems fine, but _handle_powering_off_end never run oO

Reply:
The action function has to accept an argument, cause the time will be passed as an argument to the function when it is called. If you don't plan to use that argument, you can accept arbitrary arguments.
```py
def _handle_powering_off_end(self, *args):
    ...
```
Also, you have to separate utc time from local time, depending on which function you want to use. There are two different function for tracking time, `track_point_in_time` and `track_point_in_utc_time`. So if you use the former, you should capture the current time point in local time, `now()`.

Reply:
Thanks for your reply, it work after adding a second argument to my function.

In the end I implemented a simpler solution, I save the point in time at which occur the shutdown command:
`self._end_of_power_off_command  = dt_util.utcnow() + timedelta(seconds=15)`
And defined a function to check if we are still in a powering off phase:
```
    def _power_off_in_progress(self):
        return self._end_of_power_off_command is not None and \
               self._end_of_power_off_command > dt_util.utcnow()
```

I hope it's fine like that :smile:  (it's tested and working).